test/Splitter.t.sol

```diff
@@ -0,0 +1,433 @@
+        uint256 prevRewardRate = farm.rewardRate();
+        uint256 farmLeftover = block.timestamp >= farm.periodFinish() ? 0 : farm.rewardRate() * (farm.periodFinish() - block.timestamp);
+        uint256 farmReward = vow.bump() * (WAD - splitter.burn()) / RAD;
+        uint256 prevLastUpdateTime = farm.lastUpdateTime();
+
+        vm.expectEmit(false, false, false, false); // only check event signature (topic 0)
+        emit Kick(0, 0);
+        vow.flap();
+
+        assertEq(vat.dai(address(vow)), initialVowVatDai - vow.bump());
+        assertEq(vat.dai(DAI_JOIN), initialDaiJoinVatDai + vow.bump());
+        assertEq(vat.dai(address(splitter)), 0);
+
+        assertEq(GemLike(DAI).balanceOf(UNIV2_DAI_MKR_PAIR), initialReserveDai + vow.bump() * splitter.burn() / RAD);
+        assertLt(GemLike(MKR).balanceOf(UNIV2_DAI_MKR_PAIR), initialReserveMkr);
+        assertGt(GemLike(MKR).balanceOf(address(PAUSE_PROXY)), initialMkr);
+
+        assertEq(GemLike(DAI).balanceOf(address(farm)), initialFarmDai + farmReward);
+        assertEq(farm.rewardRate(), splitter.burn() == WAD ? prevRewardRate : (farmLeftover + farmReward) / farm.rewardsDuration());
+        assertEq(farm.lastUpdateTime(), splitter.burn() == WAD ? prevLastUpdateTime : block.timestamp); 
+    }
+
+    function testConstructor() public {
+        vm.expectEmit(true, true, true, true);
+        emit Rely(address(this));
+        Splitter s = new Splitter(DAI_JOIN, address(0xfff));
+
+        assertEq(address(s.daiJoin()),  DAI_JOIN);
+        assertEq(address(s.vat()), address(vat));
+        assertEq(address(s.farm()), address(0xfff));
+        assertEq(s.wards(address(this)), 1);
+    }
+
+    function testAuth() public {
+        checkAuth(address(splitter), "Splitter");
+    }
+
+    function testAuthModifiers() public virtual {
+        assert(splitter.wards(address(this)) == 0);
+
+        checkModifier(address(splitter), string(abi.encodePacked("Splitter", "/not-authorized")), [
+            Splitter.kick.selector,
+            Splitter.cage.selector
+        ]);
+    }
+
+    function testFileUint() public {
+        checkFileUint(address(splitter), "Splitter", ["burn"]);
+    }
+
+    function testFileAddress() public {
+        checkFileAddress(address(splitter), "Splitter", ["flapper"]);
+    }
+
+    function testVatCanAfterFile() public {
+        assertEq(vat.can(address(splitter), address(0xf1)), 0);
+
+        vm.prank(PAUSE_PROXY); splitter.file("flapper", address(0xf1));
+
+        assertEq(vat.can(address(splitter), address(0xf1)), 1);
+        assertEq(vat.can(address(splitter), address(0xf2)), 0);
+
+        vm.prank(PAUSE_PROXY); splitter.file("flapper", address(0xf2));
+
+        assertEq(vat.can(address(splitter), address(0xf1)), 0);
+        assertEq(vat.can(address(splitter), address(0xf2)), 1);
+    }
+
+    function testKick() public {
+        doKick();
+    }
+
+    function testKickBurnOnly() public {
```

Comment:
Shouldn't this test just be the config of burn value and then doKick?

Reply:
You're right, fixed.